packages/infra/lib/surescripts-stack.ts

```diff
@@ -0,0 +1,527 @@
+import { Duration, NestedStack, NestedStackProps } from "aws-cdk-lib";
+import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { Function as Lambda } from "aws-cdk-lib/aws-lambda";
+import { SqsEventSource } from "aws-cdk-lib/aws-lambda-event-sources";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import { Queue } from "aws-cdk-lib/aws-sqs";
+import { Construct } from "constructs";
+import { EnvConfig } from "../config/env-config";
+import { EnvType } from "./env-type";
+import { createLambda } from "./shared/lambda";
+import { LambdaLayers } from "./shared/lambda-layers";
+import { LambdaSettings, QueueAndLambdaSettings } from "./shared/settings";
+import { createQueue } from "./shared/sqs";
+
+const connectSftpTimeout = Duration.seconds(30);
+const synchronizeSftpTimeout = Duration.minutes(15);
+const sendPatientRequestLambdaTimeout = Duration.minutes(12);
+const receiveVerificationLambdaTimeout = Duration.minutes(5);
+const receiveFlatFileResponseLambdaTimeout = Duration.minutes(15);
+
+interface SurescriptsSettings {
+  connectSftp: LambdaSettings;
+  synchronizeSftp: QueueAndLambdaSettings;
+  sendPatientRequest: QueueAndLambdaSettings;
+  receiveVerificationResponse: QueueAndLambdaSettings;
+  receiveFlatFileResponse: QueueAndLambdaSettings;
+}
+
+const settings: SurescriptsSettings = {
+  connectSftp: {
+    name: "SurescriptsConnectSftp",
+    entry: "surescripts-sftp-connect",
+    lambda: {
+      memory: 512,
+      timeout: connectSftpTimeout,
+    },
+  },
+  synchronizeSftp: {
+    name: "SurescriptsSynchronizeSftp",
+    entry: "surescripts-sftp-synchronize",
+    lambda: {
+      memory: 2048,
```

Comment:
Do we expect to need that many RAM on this one? Large files?

Reply:
I actually wasn't sure so I picked something on the safer end. Can change to your suggested value.

Reply:
It depends on how much data we're expecting to process on each invocation, or if we need a lot of CPU (it grows w/ RAM).

What's the max payload size you expect we'll process? Many transformations?

NABD, but could start w/ 512-1024 and grow as needed, if you think it should suffice.